src/NServiceBus.RavenDB/Internal/DocumentStoreManager.cs

```diff
@@ -0,0 +1,190 @@
+﻿namespace NServiceBus.RavenDB.Internal
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Configuration;
+    using System.Linq;
+    using NServiceBus.Logging;
+    using NServiceBus.Persistence;
+    using NServiceBus.Settings;
+    using Raven.Client;
+    using Raven.Client.Document;
+
+    static class DocumentStoreManager
+    {
+        static readonly ILog Logger = LogManager.GetLogger(typeof(DocumentStoreManager));
+        const string settingsKey = "RavenDbDocumentStore";
```

Comment:
could we rename `settingsKey` to something more meaningful in this context? such as `defaultDocStoreKey` or similar. While reading through the class it is hard to understand, given the way it is used, the meaning of `settingsKey`.